winch/codegen/src/isa/x64/asm.rs

```diff
@@ -1455,35 +1457,41 @@ impl Assembler {
     /// Requires `has_lzcnt` flag.
     pub fn lzcnt(&mut self, src: Reg, dst: WritableReg, size: OperandSize) {
         assert!(self.isa_flags.has_lzcnt(), "Requires has_lzcnt flag");
-        self.emit(Inst::UnaryRmR {
-            size: size.into(),
-            op: args::UnaryRmROpcode::Lzcnt,
-            src: src.into(),
-            dst: dst.map(Into::into),
-        });
+        let dst = WritableGpr::from_reg(dst.to_reg().into());
```

Comment:
I was struggling with this today with multiplication: is there no better way?

Reply:
Oh heh I was just copying from code you wrote, so I alas don't have great ideas here :(

"fixing" this would probably involve something more invasive in Winch to have typed registers throughout the entire implementation rather than just the low-level assembler would be my guess, but that's a pretty big change